### PR TITLE
feat: add signsgd memory-efficient optimizers

### PR DESCRIPTION
**Pull request by samsja**
## Summary
- Add SignAdamW optimizer that uses sign of gradients instead of storing momentum/variance state

---

> [!NOTE]
> **Medium Risk**
> Adds a new optimizer implementation and wires it into trainer config/selection, which can change training dynamics when enabled and may affect checkpoint/resume behavior for runs using it.
> 
> **Overview**
> Adds a new memory-light optimizer option, `sign_sgd`, that updates parameters using only the sign of gradients (with optional decoupled weight decay) and maintains no per-parameter momentum/variance state.
> 
> Extends the trainer optimizer config union (`OptimizerConfig`) and optimizer factory (`_create_optimizer` in `trainer/optim.py`) to recognize `type: "sign_sgd"` and construct the new `SignSGD` optimizer.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit d27d627ce875423cc4abb91fb65f02db80239e10. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/prime_rl/trainer/sign_sgd.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                loss = closure()
+
+        for group in self.param_groups:
+            for p in group["params"]:
+                if p.grad is None:
+                    continue
+
+                grad = p.grad
+                if grad.is_sparse:
+                    raise RuntimeError("SignSGD does not support sparse gradients")
```
Comment (samsja) [inline src/prime_rl/trainer/sign_sgd.py]: lest remove this

**Review by faresobeid — APPROVED** [review]
